StoreResourceLoader.js

```diff
@@ -0,0 +1,50 @@
+const clownface = require('clownface')
+const ns = require('@tpluscode/rdf-ns-builders')
+const { fromStream } = require('rdf-dataset-ext')
+const rdf = { ...require('@rdfjs/data-model'), ...require('@rdfjs/dataset') }
+const TermSet = require('@rdfjs/term-set')
+
+class StoreResourceLoader {
+  constructor ({ store }) {
+    this.store = store
+  }
+
+  async load (term) {
+    const dataset = await fromStream(rdf.dataset(), this.store.match(null, null, null, term))
+
+    if (dataset.size === 0) {
+      return null
+    }
+
+    const types = new TermSet(clownface({ dataset, term }).out(ns.rdf.type).terms)
+
+    return {
+      term,
+      dataset,
+      types
+    }
+  }
+
+  async byClass (term) {
+    const resource = await this.load(term)
+
+    return resource ? [resource] : []
+  }
+
+  async byProperty (term) {
```

Comment:
Consequently this could become

```suggestion
  async byObject (term) {
```

where the logic is that the lookup analogous to 

```
DESCRIBE ?s WHERE {
  ?s ?p ${term}
}
```

Hence, `term` is the object

Reply:
Same here, but it's a property operation: `forPropertyOperation`.